terralab/utils.py

```diff
@@ -248,3 +257,27 @@ def format_timestamp(timestamp_string: str | None) -> str:
         local_timezone
     )
     return datetime_obj.strftime("%Y-%m-%d %H:%M:%S %Z")
+
+
+# This is a workaround to make the retry messages from urllib3 more user-friendly.
+# Previously they would show up as ugly errors, i.e.
+#   Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None))
+#   after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection
+#   object at 0x102c47ef0>: Failed to establish a new connection: [Errno 61]
+#   Connection refused')': /api/pipelineruns/v1/pipelineruns?limit=10
+
+
+class RetryMessageFilter(logging.Filter):
```

Comment:
We could also suppress these warnings entirely and just return the error, but I do think there is some small UX value of letting the user know that their request was retried

Reply:
I think the final error message that says `Maximum retries reached...` is probably enough infromation on its own to let teh user know we retried w/e underlying requests.  I feel like warning "retrying" messages should maybe be presented in some "debug" mode vs at all times.  We should chat faces on this.

I might also move this to the log.py file as its essentially affects the logging of the system.